awx/main/access.py

```diff
@@ -2716,6 +2717,17 @@ class RoleAccess(BaseAccess):
     '''
 
     model = Role
+    prefetch_related = ('content_type',)
+
+    def filtered_queryset(self):
+        result = Role.visible_roles(self.user)
+        # Sanity check: is the requesting user an orphaned non-admin/auditor?
+        # if yes, make system admin/auditor mandatorily visible.
+        if not self.user.organizations.exists() and not self.user.is_superuser and not self.user.is_system_auditor:
```

Comment:
I'd suggest rearranging this, to avoid the db query for the user organizations under the superuser/auditor case.  But it's a pretty minor optimization most likely.

Reply:
I agree, although I would add that `is_system_auditor` incurs a query most of the time, but it still has caching, whereas the first term here doesn't.